shared/runtime/pyexec.c

```diff
@@ -116,7 +131,9 @@ static int parse_compile_execute(const void *source, mp_parse_input_kind_t input
         #if MICROPY_REPL_INFO
         start = mp_hal_ticks_ms();
         #endif
-        mp_call_function_0(module_fun);
+        if (!(exec_flags & EXEC_FLAG_COMPILE_ONLY)) {
```

Comment:
This compile-only option doesn't work correctly at the REPL, eg:
```
$ ./build-standard/micropython -X compile-only
MicroPython v1.26.0-preview.282.gdcbdeb839 on 2025-06-27; linux [GCC 15.1.1] version
Type "help()" for more information.
>>> 1/2
0.5
>>> 
```
That's wrong, it shouldn't print anything as the answer.

I suggest to just keep this simple and add a compile-time option to enable this:
```c
#if MICROPY_PYEXEC_COMPILE_ONLY
if (mp_compile_only)
#endif
{
    mp_call_function_0(module_fun);
}
```
Then the unix port enables this and makes `mp_compile_only` globally available and declared in `mphalport.h`.

That will mean `pyexec_file_with_flags()` isn't needed, and ports that don't use this feature won't take a hit.

Reply:
Thanks for the suggestion, I've implemented this and pushed.